### ISSUE TITLE
Support a zero_grad() function for DDP modules

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#43177]
## 🚀 Feature
Single-device modules have a `zero_grad` function that zeros gradients for the model, but if there is not an optimizer lying around, this must currently be done manually for DDP models:

```
def ddp_zero_grad(net: torch.nn.parallel.DistributedDataParallel):
    for param in net.module.parameters():
        if param.grad is not None:
            param.grad.detach_()
            param.grad.zero_()
```
## Motivation

There may be cases where DDP is used without an optimizer, and this would be a useful utility to provide for those purposes.
